### ISSUE TITLE
Deletion of connection pooler incorrectly handles ResourceNotFound

### ISSUE THREAD

**Issue by frittentheke** [zalando/postgres-operator#961]
With 1.5 the new connection pooler was added. In the function to delete the corresponding resources there is a check if the error returned from the K8S API is `ResourceNotFound` (good, as we wanted to delete it anyways), see https://github.com/zalando/postgres-operator/blob/master/pkg/cluster/resources.go#L194

But it seems to me the condition wrongly checks for the opposite case - so if it NOT this error.
I prepared a PR to adjust this: https://github.com/zalando/postgres-operator/pull/960

**Comment by FxKu** [comment]
Did you actually see the error in your logs? :smiley: 
Tried to reproduce it myself but seeing the correct log messages, although the code looks indeed wrong.

**Comment by frittentheke** [comment]
@FxKu  ... well, yes and no. Actually I noticed this when working on the Finalizer PR (https://github.com/zalando/postgres-operator/pull/941).

Until my changes in that PR the Delete function was doing "fire and forget" and simply tried to delete things once and then was done with them, success or not. With the introduction of the Finalizer I wanted to make the Operator to ensure the managed resources are really all gone before the Postgresql CR is deleted (or rather the Finalizer is removed).

I added the same methodology to handling RessourceNotFound to other resources as well - see: https://github.com/zalando/postgres-operator/pull/941/commits/05fdbb21abbaf3d0e202384c9496a37197f20ef9, that is when I noticed the issue.
